firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -2383,20 +2383,31 @@ extension BrowserViewController: SearchViewControllerDelegate {
     }
 
     func searchViewControllerWillHide(_ searchViewController: SearchViewController) {
-        guard searchEngagementState == .engaged else { return }
-        let visibleSuggestionsTelemetryInfo = searchViewController.visibleSuggestionsTelemetryInfo
-        for info in visibleSuggestionsTelemetryInfo {
-            trackEngagedSearchTelemetry(visibleSuggestionInfo: info)
+        switch searchEngagementState {
```

Comment:
I tried to DRY this out with something like:

```swift
var method: TelemetryWrapper.EventMethod
switch searchEngagementState {
case .engaged:
    method = .engagement

case .abandoned:
    method = .abandonment

default:
    return
}

let visibleSuggestionsTelemetryInfo = searchViewController.visibleSuggestionsTelemetryInfo
visibleSuggestionsTelemetryInfo.forEach { trackVisibleSuggestion(telemetryInfo: $0) }
TelemetryWrapper.gleanRecordEvent(category: .action, method: method, object: .locationBar)
```

...But I think that actually looks clunkier than duplicating the two extra lines in the cases. I'd be happy to change it if you prefer it, though!

Reply:
yeh.... I DRY at 3+. Two repetitions are whatever. lol